packages/template-retail-react-app/app/hooks/use-wish-list.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useProductLists} from '../contexts'
+import {useProducts} from 'commerce-sdk-react-preview'
+
+export const useWishList = ({listId = '', shouldFetchProductDetail = false} = {}) => {
+    const {
+        productLists,
+        error: productListsError,
+        isLoading: isProductListsLoading
+    } = useProductLists()
+    const wishLists = productLists?.data?.filter((list) => list.type === 'wish_list')
+    const currentWishlist =
+        wishLists?.data?.find((list) => list.id === listId) || wishLists?.data?.[0]
+
+    const productIds =
+        currentWishlist?.customerProductListItems?.map(({productId}) => productId).join(',') ?? ''
+
+    const {
+        data: products,
+        isLoading: isProductsLoading,
+        error: productsError
+    } = useProducts(
+        {
+            parameters: {
+                ids: productIds,
+                allImages: true
+            }
+        },
+        {
+            enabled: shouldFetchProductDetail && Boolean(productIds),
+            select: (result) => {
+                // Convert array into key/value object with key is the product id
+                return result?.data?.reduce((result, item) => {
+                    const key = item.id
+                    result[key] = item
+                    return result
+                }, {})
+            }
+        }
+    )
+    return {
+        error: productListsError || productsError,
+        // isProductListsLoading is always true if shouldFetchProductDetail is false
+        isLoading: shouldFetchProductDetail
+            ? isProductListsLoading || isProductsLoading
+            : isProductListsLoading,
```

Comment:
```suggestion
        isLoading: fetchingProductDetail
            ? isProductListsLoading || isProductsLoading
            : isProductListsLoading || isProductsInitialLoading,
```

And then keep track of this somewhere: `const fetchingProductDetail = shouldFetchProductDetail && Boolean(productIds)`

`isProductsInitialLoading` is a new variable that comes from `{isInitialLoading} = useProducts()`. I think the _initial_ loading is what you're looking for, whenever you disable a query. See the doc at https://tanstack.com/query/v4/docs/react/guides/disabling-queries#isinitialloading